controller/src/enhancements/bomb.rs

```diff
@@ -179,63 +253,144 @@ impl Enhancement for BombInfo {
+                let mut green = [0.11, 0.79, 0.26, 1.0]; // Green
+                let mut red = [0.79, 0.11, 0.11, 1.0]; // Red
+
+                // Reset Colors
+                if !ctx.settings.bomb_timer_color {
+                    orange = white;
+                    green = white;
+                    red = white;
+                }
 
-        let line_count = match &bomb_info.state {
-            C4State::Active { .. } => 3,
-            C4State::Defused | C4State::Detonated => 2,
-        };
-        let text_height = ui.text_line_height_with_spacing() * line_count as f32;
-
-        /* align to be on the right side after the players */
-        let offset_x = ui.io().display_size[0] * 1730.0 / 2560.0;
-        let offset_y = ui.io().display_size[1] * PLAYER_AVATAR_TOP_OFFSET;
-        let offset_y = offset_y
-            + 0_f32.max((ui.io().display_size[1] * PLAYER_AVATAR_SIZE - text_height) / 2.0);
-
-        ui.set_cursor_pos([offset_x, offset_y]);
-        ui.text(&format!(
-            "Bomb planted {}",
-            if bomb_info.bomb_site == 0 { "A" } else { "B" }
-        ));
-
-        match &bomb_info.state {
-            C4State::Active {
-                time_detonation,
-                defuse,
-            } => {
-                ui.set_cursor_pos_x(offset_x);
-                ui.text(&format!("Time: {:.3}", time_detonation));
-                if let Some(defuse) = defuse.as_ref() {
-                    let color = if defuse.time_remaining > *time_detonation {
-                        [0.79, 0.11, 0.11, 1.0]
-                    } else {
-                        [0.11, 0.79, 0.26, 1.0]
-                    };
-
-                    ui.set_cursor_pos_x(offset_x);
-                    ui.text_colored(
-                        color,
-                        &format!(
-                            "Defused in {:.3} by {}",
-                            defuse.time_remaining, defuse.player_name
-                        ),
+                // Helper Function
+                fn helper_text_color(
+                    ctx: &RenderContext,
+                    player_team: &u8,
+                    color_t: impl Into<Vector4<f32>>,
+                    text_t: &str,
+                    color_ct: impl Into<Vector4<f32>>,
+                    text_ct: &str,
+                ) {
+                    match &player_team {
+                        2 => {
+                            // Terrorists
+                            ctx.ui.text_colored(color_t, text_t)
+                        }
+                        3 => {
+                            // Counter-Terrorists
+                            ctx.ui.text_colored(color_ct, text_ct)
+                        }
+                        &_ => {
+                            log::warn!("weird team id! {}", &player_team)
+                        }
+                    }
+                }
+
+                if !matches!(&bomb_info.state, C4State::Dropped) {
```

Comment:
Why don't you use the general match statement bellow?

Reply:
I would have to put it in Active, Defused, Detonated then, this is the least duplicate code